File: bowling.py
```
def bowling_score(rolls):
    score = 0
    roll_index = 0
    for frame in range(1, 11):
        if rolls[roll_index] == 'X':
            score += 10
            roll_index += 1
        elif '/' in rolls[roll_index:roll_index+2]:
            first_roll, second_roll = map(int, rolls[roll_index:roll_index+2].replace('/', '').split())
            if first_roll + second_roll == 10:
                score += 10
            else:
                score += first_roll + second_roll
            roll_index += 2
        else:
            first_roll = int(rolls[roll_index])
            if first_roll == 10:
                score += 10
                roll_index += 1
            else:
                score += first_roll
                if rolls[roll_index+1] != 'X' and rolls[roll_index+1] != '/':
                    second_roll = int(rolls[roll_index+1])
                    score += second_roll
                roll_index += 1
    return score

File: bowling.cpp
#include <iostream>
#include <string>

int score(std::string s) {
    int total = 0;
    int frame = 1;
    int i = 0;

    while (frame <= 10) {
        if (s[i] == 'X') {
            total += 10;
            total += (s[i + 2] == 'X') ? 10 : (s[i + 2] == '/') ? 10 - s[i + 1] + 10 : (s[i + 1] == '/') ? 10 : s[i + 1] - '0';
            i++;
        } else if (s[i + 1] == '/') {
            total += 10;
            total += (s[i + 2] == 'X') ? 10 : s[i + 2] - '0';
            i += 2;
        } else {
            total += (s[i] == '-') ? 0 : s[i] - '0';
            total += (s[i + 1] == '-') ? 0 : s[i + 1] - '0';
            i += 2;
        }
        frame++;
    }

    return total;
}

int main() {
    std::string s;
    std::cin >> s;
    std::cout << score(s) << std::endl;
    return 0;
}